src/coffee/controllers.coffee

```diff
@@ -44,6 +44,174 @@ define ['angular', 'lodash', 'angular-cookies', 'services'], (ng, L) ->
 
     historyListener.watch scope
 
+  #############################################
```

Comment:
I think this controller is large enough to justify extraction to its own file. See HistoryCtrl et. al. for examples.